src/main/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModification.java

```diff
@@ -231,116 +231,124 @@ private void processPhaseTapChanger(Network network,
+                                                 AttributeModification<Double> regulationValueModification,
+                                                 AttributeModification<Double> targetDeadbandModification,
                                                  List<ReportNode> regulationReports) throws NetworkModificationException {
-        AttributeModification<PhaseTapChanger.RegulationMode> finalRegulationMode = regulationModeValue;
-        AttributeModification<Double> finalTargetDeadband = modifyTargetDeadband;
-        AttributeModification<Boolean> finalRegulating = null;
-
-        // Regulation Mode is linked to regulating
-        if (isModification) {
-            // modification
-            if (regulationModeValue != null) {
-                Double regulationValue = modifyRegulationValue == null ? null : modifyRegulationValue.getValue();
+        AttributeModification<PhaseTapChanger.RegulationMode> finalRegulationModeModification = regulationModeModification;
+        AttributeModification<Double> finalTargetDeadbandModification = targetDeadbandModification;
+        AttributeModification<Boolean> finalRegulatingModification = null;
+
+        // ---  Regulation Mode impacts on regulating => pre-processing regulating --- //
+        if (isModification) { // modifying an existing extension
+            // modification check
+            if (regulationModeModification != null) {
                 PhaseTapChanger.RegulationMode oldRegulationMode = phaseTapChanger.getRegulationMode();
-                PhaseTapChanger.RegulationMode newRegulationMode = regulationModeValue.getValue();
-                double targetDeadBand = phaseTapChanger.getTargetDeadband();
-                // if previous regulation mode was FIXED_TAP there is a check that there are all the attributes needed to regulate
+                PhaseTapChanger.RegulationMode newRegulationMode = regulationModeModification.getValue();
+                double oldTargetDeadBand = phaseTapChanger.getTargetDeadband();
+
                 if (oldRegulationMode == PhaseTapChanger.RegulationMode.FIXED_TAP) {
-                    if (regulationValue == null) {
-                        throw new NetworkModificationException(MODIFY_TWO_WINDINGS_TRANSFORMER_ERROR, "Regulation value can not be null if phase tap changer is now regulating");
-                    }
-                    finalRegulating = AttributeModification.toAttributeModification(true, OperationType.SET);
-                    finalRegulationMode = AttributeModification.toAttributeModification(newRegulationMode, OperationType.SET);
-                    if (Double.isNaN(targetDeadBand) && modifyTargetDeadband == null) {
-                        finalTargetDeadband = AttributeModification.toAttributeModification(0.0, OperationType.SET);
+                    // if new regulation mode is FIXED_TAP set regulating to false
+                    if (newRegulationMode == PhaseTapChanger.RegulationMode.FIXED_TAP) {
```

Comment:
I dont think it is possible to modify the regulation mode to change from FIXED_TAP to FIXED_TAP is it ?

Reply:
if it is I understand